sympy/physics/control/lti.py

```diff
@@ -135,34 +184,118 @@ class TransferFunction(Basic, EvalfMixin):
     resultant transfer function using ``.doit()`` method or by ``.rewrite(TransferFunction)``.
 
     >>> ((tf9 + tf10) * tf12).doit()
-    TransferFunction((1 - s)*((-p + s)*(s**2 + s + 1) + (s + 1)*(s + 3)), (s + 3)*(s**2 + 4)*(s**2 + s + 1), s)
+    TransferFunction(p*s**3 - p - s**4 - s**3 - 3*s**2 + 2*s + 3, s**5 + 4*s**4 + 8*s**3 + 19*s**2 + 16*s + 12, s)
     >>> (tf9 * tf10 - tf11 * tf12).rewrite(TransferFunction)
-    TransferFunction(-(1 - s)*(s + 3)*(s**2 + s + 1)*(4*s**2 + 2*s - 4) + (-p + s)*(s - 1)*(s + 1)*(s**2 + 4), (s - 1)*(s + 3)*(s**2 + 4)*(s**2 + s + 1), s)
+    TransferFunction(-p*s**4 - 3*p*s**2 + 4*p + 4*s**6 + 15*s**5 + 2*s**4 - 13*s**3 - 14*s**2 - 6*s + 12, s**6 + 3*s**5 + 4*s**4 + 11*s**3 - 3*s**2 - 4*s - 12, s)
 
     See Also
     ========
 
-    Feedback, Series, Parallel
+    TransferFunctionMatrix, Feedback, Series, Parallel
+
+    References
+    ==========
+
+    .. [1] https://en.wikipedia.org/wiki/Transfer_function
+    .. [2] https://en.wikipedia.org/wiki/Laplace_transform
 
     """
     def __new__(cls, num, den, var):
         num, den = _sympify(num), _sympify(den)
+        num, den = expand(num), expand(den)
 
         if not isinstance(var, Symbol):
             raise TypeError("Variable input must be a Symbol.")
+
         if den == 0:
-            raise ValueError("TransferFunction can't have a zero denominator.")
-
-        if (((isinstance(num, Expr) and num.has(Symbol) and not num.has(exp)) or num.is_number) and
-            ((isinstance(den, Expr) and den.has(Symbol) and not den.has(exp)) or den.is_number)):
-                obj = super().__new__(cls, num, den, var)
-                obj._num = num
-                obj._den = den
-                obj._var = var
-                return obj
+            raise ZeroDivisionError("TransferFunction can't have a zero denominator.")
+
+        if (((isinstance(num, Expr) and not isinstance(num, ImmutableMatrix) and num.has(Symbol) and not num.has(exp)) or num.is_number) and
+            ((isinstance(den, Expr) and not isinstance(num, ImmutableMatrix) and den.has(Symbol) and not den.has(exp)) or den.is_number)):
+            obj = super(TransferFunction, cls).__new__(cls, num, den, var)
+            obj._num = num
+            obj._den = den
+            obj._var = var
+            obj._num_inputs, obj._num_outputs = 1, 1 # TODO: Remove _num_inputs and _num_outputs for TF and everywhere else except TFM
+            return obj
+
         else:
             raise TypeError("Unsupported type for numerator or denominator of TransferFunction.")
 
+    @classmethod
+    def from_rational_expression(cls, expr, var=None):
```

Comment:
@oscarbenjamin @asmeurer, does sympy allow such classmethods in the codebase?

Reply:
I see several similar methods in this file:
https://github.com/sympy/sympy/blob/21aa3ab7111292aad003827164639716c84c5294/sympy/polys/matrices/domainmatrix.py#L215